src/main/java/com/vesperin/base/EclipseJavaParser.java

```diff
@@ -38,6 +38,35 @@ public EclipseJavaParser(){
     astParser.setCompilerOptions(options);
   }
 
+  public void setASTParserEnv(String[] classpathEntries, String[] sourcepathEntries, String[] encodings) {
+      this.astParser.setEnvironment(classpathEntries, sourcepathEntries, encodings, true);
+  }
+
+  @Override
+    public Context parseJava(Source code) {
+        this.astParser.setKind(ASTParser.K_COMPILATION_UNIT);
+        this.astParser.setStatementsRecovery(true);
+        this.astParser.setBindingsRecovery(true);
+        this.astParser.setUnitName(code.getName() + JAVA_EXTENSION);
+        this.astParser.setSource(code.getContent().toCharArray());
+
+        ASTNode unit = null;
+        ParsedUnit parsedUnit = null;
+        try {
+            unit = this.astParser.createAST(null);
+            if(unit == null){
+                parsedUnit = ParsedUnit.empty();
+            }
+
+            parsedUnit = ParsedUnit.makeUnit(unit, false );
```

Comment:
this breaks a contract between `mix` and some other Vesperin's tools, which care whether you are parsing expressions or simply statements

Reply:
Yes, I will remove this method. Thanks for the explanation.